subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -1292,6 +1316,9 @@ int nrf_cloud_rest_send_device_message(struct nrf_cloud_rest_context *const rest
 	if (req.body) {
 		cJSON_free((void *)req.body);
 	}
+	if (d2c) {
+		k_free(d2c);
+	}
 	cJSON_Delete(root_obj);
```

Comment:
Since you call cJSON_AddRawToObjectCS with json_msg, is it possible that this delete call will recursively delete the passed in json_msg before the caller is expecting it? Could this lead to a double deletion? Or are these objects reference counted by cJSON?

I'm not 100% certain on how cJSON memory management is structured.

Reply:
I see some mention of reference counting in some cJSON sources, but I don't know whether AddRawToObjectCS will increment the reference count, nor if cJSON_Delete even respects the reference count.

However, if you find that deletion/addition do not increment/decrement some reference counter, this presents itself as a potential solution:
https://cjson.docsforge.com/master/api/cJSON_CreateObjectReference/

It's also totally possible that I'm just misreading your code, and that this isn't a problem to begin with

Reply:
Oh, yeah, I totally misunderstand. Passed in json_msg is a raw char*, and the cJSON_AddRawToObjectCS function just creates a reference.

This is a non-issue